### ISSUE TITLE
Is language list deprecated? What will replace it?

### ISSUE THREAD

**Issue by dhamaris** [ec-europa/europa-component-library#2915]
I've seen that Language selector appears under "Deprecated". What will replace it?
https://ec.europa.eu/component-library/v3.8.2/ec/deprecated/language-list/usage/

Thank you

**Comment by planctus** [comment]
Hello @dhamaris, the language switcher as a standalone component is deprecated and it will disappear in v4.
The new language switcher is already in place, in the new site header component:
https://ec.europa.eu/component-library/playground/ec/?path=/story/components-site-wide-site-header--core
If you use the twig templates, there is a [template](https://github.com/ec-europa/europa-component-library/blob/v3-dev/src/implementations/twig/components/site-header/site-header-language-switcher.html.twig) meant to render the markup of this new language switcher.
Thanks!

**Comment by dhamaris** [comment]
Thank you for the information! 
I am using twig templates for our drupal systems, but I am not sure of how can i use the [template](https://github.com/ec-europa/europa-component-library/blob/v3-dev/src/implementations/twig/components/site-header/site-header-language-switcher.html.twig) you provide, I just copy the markup from the playground.

**Comment by planctus** [comment]
why, @dhamaris..? The twig templates are meant for you not to have to reproduce all the complex markup of our components, copying from storybook it's not really recommended.
The template is included by the site header template:
https://github.com/ec-europa/europa-component-library/blob/v3-dev/src/implementations/twig/components/site-header/site-header.html.twig

So for you it would just be a matter of passing the right data, including the data for the language switcher.
What is useful in our playground, is to take a look at the "notes" tab, there you find an example include code for the twig template that is reporting the data you are currently seeing in the demo:
This is what you find in that tab for the site-header:

`{% include '@ecl/site-header/site-header.html.twig' with 
{
logo: {
title: "European Commission",
alt: "European Commission logo",
path: "/component-library/example#xtxxn",
language: "en",
src_desktop: "static/media/logo-ec--en.65cfd447.svg"
},
language_selector: {
href: "/component-library/example#v4aks",
label: "English",
aria_label: "Change language, current language is English",
code: "en",
overlay: {
title: "Select your language",
items: [
{
label: "български",
path: "/component-library/example#ojp98",
lang: "bg"
},
{
label: "español",
path: "/component-library/example#yv0v0",
lang: "es"
},
{
label: "čeština",
path: "/component-library/example#jgtoz",
lang: "cs"
},
{
label: "dansk",
path: "/component-library/example#yao92",
lang: "da"
},
{
label: "Deutsch",
path: "/component-library/example#slq1w",
lang: "de"
},
{
label: "eesti",
path: "/component-library/example#ajnam",
lang: "et"
},
{
label: "ελληνικά",
path: "/component-library/example#9iwll",
lang: "el"
},
{
label: "English",
path: "/component-library/example#ub8uo",
active: true,
lang: "en"
},
{
label: "français",
path: "/component-library/example#cyz4y",
lang: "fr"
},
{
label: "Gaeilge",
path: "/component-library/example#44tbf",
lang: "ga"
},
{
label: "hrvatski",
path: "/component-library/example#jz7du",
lang: "hr"
},
{
label: "italiano",
path: "/component-library/example#yufei",
lang: "it"
},
{
label: "latviešu",
path: "/component-library/example#tvxh1",
lang: "lv"
},
{
label: "lietuvių",
path: "/component-library/example#ouqay",
lang: "lt"
},
{
label: "magyar",
path: "/component-library/example#bmb23",
lang: "hu"
},
{
label: "Malti",
path: "/component-library/example#vzz3h",
lang: "mt"
},
{
label: "Nederlands",
path: "/component-library/example#p5yth",
lang: "nl"
},
{
label: "polski",
path: "/component-library/example#2x1ge",
lang: "pl"
},
{
label: "português",
path: "/component-library/example#9okxw",
lang: "pt"
},
{
label: "română",
path: "/component-library/example#uo36s",
lang: "ro"
},
{
label: "slovenčina",
path: "/component-library/example#f928r",
lang: "sk"
},
{
label: "slovenščina",
path: "/component-library/example#wcx3u",
lang: "sl"
},
{
label: "suomi",
path: "/component-library/example#yyy3x",
lang: "fi"
},
{
label: "svenska",
path: "/component-library/example#424fz",
lang: "sv"
}
],
close_label: "Close",
non_eu_items: [
{
label: "中文",
path: "/component-library/example#71hq3",
lang: "zh"
},
{
label: "Türk",
path: "/component-library/example#7nycd",
lang: "tr"
},
{
label: "pусский",
path: "/component-library/example#qdthi",
lang: "ru"
},
{
label: "Català",
path: "/component-library/example#6a0do",
lang: "ca"
},
{
label: "عَرَبِيّ",
path: "/component-library/example#7doch",
lang: "ar"
}
]
},
eu_category: "Official EU languages:",
non_eu_category: "Other languages:"
},
search_toggle: {
label: "Search",
href: "/component-library/example#wimax"
},
search_form: {
text_input: {
id: "search-input-id",
label: "Search",
placeholder: "Placeholder text"
},
button: {
label: "Search"
},
extra_attributes: [
{
name: "id",
value: "search-form-id"
}
]
},
icon_file_path: "...path to youtr icons sprite",
has_menu: false,
site_name: ""
} %}` for the site header core. As you see it contains all the data that the site header template will then pass to the language switcher.
Hope this helps ;)

**Comment by dhamaris** [comment]
Oh I see, I didn't know that you had templates for that, i was doing extra work, my bad...
Thank you!

**Comment by dhamaris** [comment]
Hello @planctus, sorry to reopen this ticket but I don't understand how can i resolve all @ecl references on the template you used as an example: 
https://github.com/ec-europa/europa-component-library/blob/v3-dev/src/implementations/twig/components/site-header/site-header.html.twig
I get:

Twig\Error\LoaderError: Template "@ecl/picture/picture.html.twig" is not defined in "@ecl_theme/includes/site-header.html.twig" at line 233. in Twig\Loader\ChainLoader->getCacheKey() (line 99 of /opt/app-root/src/vendor/twig/twig/src/Loader/ChainLoader.php) [@uuid].

How can i get this @ecl folder with all these twigs? I am working with composer and drupal.

**Comment by planctus** [comment]
hello @dhamaris, as you know we work with npm packages, mainly, all the templates can be found in a single package, preset-ec or preset-eu, all of our templates use the @ecl namespace, so you might need to configure your twig environment in order to define a loader using this namespace in the path you will choose to move these templates in.
I will paste here our configuration, to give an example, but we use twing and its twing loader, in php things could be a bit different:
```
const path = require('path');
const { TwingEnvironment, TwingLoaderFilesystem } = require('twing');
const componentAbsPath = path.resolve(
  __dirname,
  '../../../implementations/twig/components',
);
const compositionsAbsPath = path.resolve(
  __dirname,
  '../../../implementations/twig/compositions',
);

const loader = new TwingLoaderFilesystem(componentAbsPath);

// In storybook we get this returned as an instance of
// TWigLoaderNull, we need to avoid processing this.
if (typeof loader.addPath === 'function') {
  // Add namespace.
  loader.addPath(componentAbsPath, 'ecl');
  loader.addPath(compositionsAbsPath, 'ecl');
}
module.exports = new TwingEnvironment(loader);
```

**Comment by planctus** [comment]
Sorry @dhamaris, i realized only now that pointing you to the preset-ec or preset-eu package to find all the templates was simply wrong, those packages distribute all the vanilla resources, css, js, icons but not the twig templates. For those i fear that we do not have any package that groups them together, so you will need to fetch all the @ecl/twig-component-{component-name} packages. To help you with those you can check the list of all the distributed packages in every recent release, this is for ecl 4.4.0: https://github.com/ec-europa/europa-component-library/releases/download/v4.4.0/ECL-npm-packages

**Comment by dhamaris** [comment]
No problem! I also realized afterwards that using the templates from [/twig/components](https://github.com/ec-europa/europa-component-library/blob/v3-dev/src/implementations/twig/components) path is not that simple because Drupal structures sections differently, so data has to be preprocess and prepared to be used here. I found an openeuropa [theme](https://github.com/openeuropa/oe_theme ) that apparently does this, but still in beta for eclv4, and in the past we had problems installing it, so I wanted to explore using the templates myself. Not that easy either!
Thank you for all the info though!
